redis/simpleresource.go

```diff
@@ -0,0 +1,86 @@
+package redis
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/datatrails/go-datatrails-common/logger"
+	otrace "github.com/opentracing/opentracing-go"
+)
+
+func NewSimpleResource(name string, cfg RedisConfig, resType string) (*SimpleResource, error) {
+	client, err := NewRedisClient(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create redis client: %w", err)
+	}
+
+	return &SimpleResource{
+		ClientContext: ClientContext{
+			cfg:  cfg,
+			name: name,
+		},
+		client:    client,
+		keyPrefix: fmt.Sprintf("%s/%s/%s", cfg.Namespace(), resType, name),
+	}, nil
+}
+
+// SimpleResource
+type SimpleResource struct {
+	ClientContext
+	client    Client
+	keyPrefix string
+}
+
+func (r *SimpleResource) URL() string {
+	return r.cfg.URL()
+}
+
+func (r *SimpleResource) Name() string {
+	return r.name
+}
+
+func (r *SimpleResource) Key(tenantID string) string {
+	return r.keyPrefix + "/" + tenantID
+}
+
+func (r *SimpleResource) Set(ctx context.Context, tenantID string, value any) error {
```

Comment:
minor: would be nice to have a pairwise unit test that mocks the redis calls, just to make sure we are marshaling/unmarshaling the json correctly.

put as minor as it works after manual testing and don't want to hold up merging

Reply:
Agree that this is a great idea, will choose to defer it if that's alright :)